### DISCUSSION TITLE
SFTP: Too many auth failures

### DISCUSSION THREAD

**Discussion by siggib007** [iterate-ch/cyberduck#15274] [Q&A] [answered]
When I try to use Cyberduck for SFTP using SSH key I get login failure "too many authentication failure" whether I specify the private key or not. I feel like it is wanting me to provide a password, which I'm leaving blank because there is no password on this account. From the command line I can connect just fine. Both of these work like a charm
>sftp siggib@htzn.supergeek.us
>sftp siggib@htzn.supergeek.us -i \user\siggib\.ssh\id_ed25519

Wondering what I'm doing wrong. Here is screenshot of the bookmark.
![image](https://github.com/iterate-ch/cyberduck/assets/20001822/ce70bb49-0a4b-408b-8751-ea3384e99228)

**Comment by ylangisc** [comment]
Can you please [enable debug logging](https://docs.cyberduck.io/cyberduck/support/#debug-logging-level) in the Preferences, recreate the issue and post the log file?

**Reply by siggib007** [reply]
Here you go
[cyberduck.log](https://github.com/iterate-ch/cyberduck/files/13206427/cyberduck.log)

**Comment by ylangisc** [comment]
Thanks @siggib007. It seems that you have an OpenSSH (or compatible) agent running with 6 identities registered. As all of them are tried with a server side auth attempt max <6 , this can lead to the server error you see.  That it works from the command line is most probably due to the different identity order tried. To be sure that it works in any circumstance (and with Cyberduck) I suggest to create an [OpenSSH config file](https://docs.cyberduck.io/protocols/sftp/#openssh-configuration-interoperability) with an entry for you server and the `IdentitiesOnly` option, i.e.
```
Host htzn.supergeek.us
	User siggib
	IdentityFile C:\Users\SiggiBjarnason\.ssh\id_ed25519
	IdentitiesOnly
```

**Comment by siggib007** [comment]
I just unregistered couple of identities from the agent and now I'm getting different errors
![image](https://github.com/iterate-ch/cyberduck/assets/20001822/99271ba6-8f9a-4f3d-8f5b-d7f3ae5efdb8)
![image](https://github.com/iterate-ch/cyberduck/assets/20001822/8be74d16-2345-45f9-ab2e-bbeb120ae399)
The weird thing is if I click the URL in the bookmarks, instead of connecting with the bookmark, I get connected just fine, only I end up at / not /home/siggib, even if that is what the URL says. Also if I close out and open up, it auto connects just fine, and I don't even have auto connect turned on :-D  Here is another log file. 
[cyberduck.log](https://github.com/iterate-ch/cyberduck/files/13209062/cyberduck.log)

**Comment by ylangisc** [comment]
Authentication seems to work fine now but there is an issue opening the default path. Different paths are tried if there isn't any set in the bookmark. I can also see that `/home/siggib` is tried but the server immediately closes the connection. As I don't have any server logs it's quite unclear what is going on. The connection that finally works is with `/`. What you can try is to issue the same command from the command line and see what happens:
```
ssh siggib@htzn.supergeek.us -i \user\siggib.ssh\id_ed25519 stat /home/siggib
```

**Comment by siggib007** [comment]
Here you go
```
C:\Users\siggib>ssh siggib@htzn.supergeek.us stat /home/siggib
  File: /home/siggib
  Size: 4096            Blocks: 8          IO Block: 4096   directory
Device: 253,3   Inode: 131         Links: 12
Access: (0755/drwxr-xr-x)  Uid: ( 1000/  siggib)   Gid: ( 1000/  siggib)
Access: 2023-10-30 18:56:10.742827036 +0000
Modify: 2023-10-30 14:36:04.768624639 +0000
Change: 2023-10-30 14:36:04.768624639 +0000
 Birth: 2022-10-30 22:38:13.802599207 +0000

C:\Users\siggib>
```
This oddity seems to be limited to my desktop. I'm not experiencing this on my laptop.

**Reply by siggib007** [reply, marked as answer]
Been playing with this some more. and I noticed that my laptop had "default" for "transfer files" UTF-8 Encoding, while my desktop had "Open Multiple connections" and "UTF-16LE" encoding. So I switched my desktop to UTF-8 encoding and now everything works. I guess my desktop had gotten the encoding messed up at some point and that was the cause.
